### ISSUE TITLE
UnhandledPromiseRejectionWarning in tests

### ISSUE THREAD

**Issue by andywhite37** [reazen/relude#101]
I was scrolling through the tests and noticed these messages:

```
(node:2602) UnhandledPromiseRejectionWarning: my error
(node:2602) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:2602) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:2602) UnhandledPromiseRejectionWarning: my error
(node:2602) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 4)
```

This issue is to just see what's happening - it's possible these are expected warnings, but they might also be unexpected.

**Comment by mlms13** [comment]
I've seen warnings like that before (outside of the context of Relude) in cases where it really seemed like the failing Promise should be caught and handled.

**Comment by andywhite37** [comment]
It will be fun when node makes unhandled promises crash the app...  so many apps will be crashing!

**Comment by andywhite37** [comment]
The two places this is raising warnings are two tests that are specifically testing for rejected promises, and the tests are in fact catching the errors and asserting the error messages...  I'm not sure what to do about this.

**Comment by mlms13** [comment]
Using Relude Fetch, I just got an `Unhandled promise rejection` error in the browser's console. It's frustrating because this promise is being handled so much better than any normal JS garbage promises would be handled. :P

Not sure what to do about this, but it might be worth looking into it, or else handled network errors are going to start producing garbage in our front-end logging.

**Comment by mlms13** [comment]
OK, I think I have an idea... from the `Js.Promise.catch` branch, we're [returning the original (failed) promise](https://github.com/reazen/relude/blob/399ce07eb7871bfe065ebf4e510b8b26e363fdfa/src/js/Relude_Js_Promise.re#L23), which tricks the runtime into thinking we haven't handled it.

By comparison, RationalJS/Future [effectively bimaps the then_/catch branches](https://github.com/RationalJS/future/blob/master/src/FutureJs.re#L16-L29) into a successful `Js.Promise.t(unit)` (which is fine, since the whole thing is ignored anyway).

**Comment by andywhite37** [comment]
Ah, that makes a lot of sense!

**Comment by andywhite37** [comment]
I went ahead and fixed this in v0.28.0 (to be released soon).
